### DISCUSSION TITLE
[android] generating java bindings

### DISCUSSION THREAD

**Discussion by cpheinrich** [google/filament#5458] [Q&A] [answered]
Hello, 

We are interested in using Filament in our project by adding additional functionality in the native layer and building from source for multiple platforms. If we add our own native code we'll need to generate java bindings to use it on Android, hence:

Are the java bindings, e.g. code in https://github.com/google/filament/tree/main/android/filament-android/src/main/cpp and their .java counterparts, generated automatically? If so, is there any documentation around the tooling and process for doing this?

**Comment by romainguy** [comment, marked as answer]
Those bindings are written by hand. IntelliJ/Android Studio can auto generate the JNI signatures from Java definitions though.
